server/ingest/handler/ingest_handler.go

```diff
@@ -19,12 +23,14 @@ import (
 // that is common to all protocols
 type IngestHandler struct {
 	indexer  indexer.Interface
+	ingester *ingest.Ingester
 	registry *registry.Registry
 }
 
-func NewIngestHandler(indexer indexer.Interface, registry *registry.Registry) *IngestHandler {
+func NewIngestHandler(indexer indexer.Interface, ingester *ingest.Ingester, registry *registry.Registry) *IngestHandler {
```

Comment:
If you want to get rid of `IndexContent` then the `indexer` argument is not needed.

Reply:
doesn't hurt to keep it piped through to this server